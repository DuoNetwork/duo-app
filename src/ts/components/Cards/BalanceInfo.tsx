import * as React from 'react';
import util from 'ts/common/util';
import { SCardAssetTag } from './_styled';

interface IProps {
	icon: string;
	name: string;
	value: number;
	mobile?: boolean;
}
export default class BalanceInfo extends React.Component<IProps> {
	public render() {
		const { icon, name, value, mobile } = this.props;
		return (
			<SCardAssetTag mobile={mobile}>
				<div className="bg-logo">
					<img src={icon} />
				</div>
				<div className="tag-title">
					<h3>{name}</h3>
				</div>
				<div className="tag-content">
					<div>
						<div>
							<div className={'tag-price'}>{util.formatBalance(value)}</div>
						</div>
					</div>
				</div>
			</SCardAssetTag>
		);
	}
}
