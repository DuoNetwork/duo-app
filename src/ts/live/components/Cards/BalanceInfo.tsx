import * as React from 'react';
import * as CST from '../../common/constants';
import contract from '../../common/contract';
import util from '../../common/util';
import { SCardAssetTag, SRefreshButton } from './_styled';

interface IProps {
	icon: string;
	name: string;
	value: number;
	allowance?: number;
	locale?: string;
	account?: string;
	mobile?: boolean;
}
export default class BalanceInfo extends React.Component<IProps> {
	public render() {
		const { icon, name, value, allowance, locale, account, mobile } = this.props;
		return (
			<SCardAssetTag mobile={mobile}>
				<div className="bg-logo">
					<img src={icon} />
				</div>
				<div className="tag-title">
					<h3>{name}</h3>
				</div>
				<div className="tag-content">
					<div>
						<div>
							<div className={'tag-price'}>{util.formatBalance(value)}</div>
							{allowance !== undefined ? (
								<div className="tag-subtext">
									{util.formatBalance(Math.min(value, allowance)) +
										' ' +
										CST.TH_ALLOWANCE[locale || CST.LOCALE_EN]}
									<SRefreshButton
										icon="plus-square-o"
										disabled={!account || account === CST.DUMMY_ADDR}
										onClick={() =>
											account &&
											contract.duoApprove(
												account,
												__KOVAN__
													? CST.CUSTODIAN_ADDR_KOVAN
													: CST.CUSTODIAN_ADDR_MAIN,
												1e8
											)
										}
									/>
								</div>
							) : null}
						</div>
					</div>
				</div>
			</SCardAssetTag>
		);
	}
}
