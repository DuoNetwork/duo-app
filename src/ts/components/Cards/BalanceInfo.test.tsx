import { shallow } from 'enzyme';
import * as React from 'react';
import BalanceInfo from './BalanceInfo';

describe('BalanceInfo Test', () => {
	describe('Test Snapshot', () => {
		const mobile = true;
		const icon = 'test';
		const name = '123';
		const value = 123;

		it('Test Snapshot', () => {
			const wrapper = shallow(
				<BalanceInfo
					icon={icon}
					name={name}
					value={value}
					mobile={mobile}
				/>
			);
			expect(wrapper).toMatchSnapshot();
			const wrapper1 = shallow(
				<BalanceInfo
					icon={icon}
					name={name}
					value={value}
					mobile={mobile}
				/>
			);
			expect(wrapper1).toMatchSnapshot();
			const wrapper2 = shallow(
				<BalanceInfo
					icon={icon}
					name={name}
					value={value}
					mobile={mobile}
				/>
			);
			expect(wrapper2).toMatchSnapshot();
		});
	});
});
