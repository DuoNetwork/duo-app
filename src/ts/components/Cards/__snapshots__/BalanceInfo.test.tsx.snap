// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthCard Test User Login Test Snapshot 1`] = `
<styled.div
  mobile={true}
>
  <div
    className="bg-logo"
  >
    <img
      src="test"
    />
  </div>
  <div
    className="tag-title"
  >
    <h3>
      123
    </h3>
  </div>
  <div
    className="tag-content"
  >
    <div>
      <div>
        <div
          className="tag-price"
        >
          123.0
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;

exports[`AuthCard Test User Login Test Snapshot 2`] = `
<styled.div
  mobile={true}
>
  <div
    className="bg-logo"
  >
    <img
      src="test"
    />
  </div>
  <div
    className="tag-title"
  >
    <h3>
      123
    </h3>
  </div>
  <div
    className="tag-content"
  >
    <div>
      <div>
        <div
          className="tag-price"
        >
          123.0
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;

exports[`AuthCard Test User Login Test Snapshot 3`] = `
<styled.div
  mobile={true}
>
  <div
    className="bg-logo"
  >
    <img
      src="test"
    />
  </div>
  <div
    className="tag-title"
  >
    <h3>
      123
    </h3>
  </div>
  <div
    className="tag-content"
  >
    <div>
      <div>
        <div
          className="tag-price"
        >
          123.0
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;
